qiskit_dynamics/backend/dynamics_backend.py

```diff
@@ -438,6 +456,29 @@ def default_experiment_result_function(
             seed=seed,
             header=QobjHeader(name=experiment_name),
         )
+    elif backend.options.meas_level == MeasLevel.KERNELED:
+        # generate IQ
+        measurement_data = _iq_data(
+            backend.options,
+            yf,
+            measurement_subsystems=measurement_subsystems,
+            seed=seed,
+        )
+
+        if backend.options.meas_return == MeasReturnType.AVERAGE:
+            measurement_data = np.average(measurement_data, axis=0)
+
```

Comment:
Does some rearranging of the measurement data using `memory_slot_indices` have to be done here?

Reply:
done